specs/dwritecore/DWriteCore.md

```diff
@@ -510,9 +510,189 @@ The following other DirectWrite APIs also use the hybrid font selection algorith
 -   `IDWriteTextLayout` if the typographic font family model is enabled by calling
     `IDWriteTextLayout4::SetFontAxisValues` or `IDWriteTextLayout4::SetAutomaticFontAxes`
 
-# Examples
-
-## Creating an isolated factory
+## Color Font Support
+
+By default, a glyph has a shape but no intrinsic color. Both DirectWrite and Direct2D have
+`DrawGlyphRun` methods that render glyph runs by filling the glyph shapes with a specified text
+color. For convenience, we will refer to this as _monochrome_ glyph rendering. All fonts have
+monochrome glyphs, but a color font also has color representations of some glyphs. To render glyphs
+in color, an application must use different glyph rendering APIs instead of calling the monochrome
+`DrawGlyphRun` method.
+
+Note that color font support does not affect text layout it at all. The text layout always produces
+monochrome glyph runs. Color is enabled at the glyph rendering level by substituting color
+representations for the monochrome glyphs.
+
+The easiest way to enable color is to call the `DrawGlyphRunWithColor` method instead of calling
```

Comment:
It's unfortunate because DrawGlyphRunWithColor sounds like it should take a color. Essentially, DrawGlyphRun and DrawGlyphRunWithColor are backwards from what you'd expect.

What about DrawColorGlyphRun? The name feels slightly more specific to the purpose and different from the confusion.

Reply:
`DrawGlyphRunWithColor` does take a color parameter. The specified color is used to render monochrome glyphs or sometimes even for parts of color glyphs (since a color glyph can reference the foreground color). The "WithColor" in the name is intended to mean "with color fonts enabled."

I thought of `DrawColorGlyphRun`, but I thought that might be confusing because there is an existing `DWRITE_COLOR_GLYPH_RUN` structure, which is returned by `IDWriteColorGlyphRunEnumerator`. Thus, `DrawColorGlyphRun` sounds like the name of a function you would call to draw each of the enumerated color runs after calling `TranslateColorGlyphRun`. That's not what this proposed function is.

Perhaps a better name would be `DrawGlyphRunWithColorFontsEnabled`? It's a bit long, but more explicit. It also nicely parallels an existing flag, `D2D1_DRAW_TEXT_OPTIONS_ENABLE_COLOR_FONT`, which can be passed to the `ID2D1RenderTarget::DrawTextLayout` method.

Reply:
Understood regarding the concerns with the existing type struct. Of all the ideas mentioned, I suppose I do prefer the name you ended up with, DrawGlyphRunWithColor

Reply:
I'll stick with DrawGlyphRunWithColor, then, unless API reviewers have a better idea.

Reply:
RECOMMEND: Make an overload-ish of DrawGlyphRun, with this including the support level parameter at the end. Long term, this becomes the right API to target.